vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -92,16 +92,49 @@ void finish_example(VW::workspace& all, automl<CMType>& data, VW::multi_ex& ec)
   VW::finish_example(all, ec);
 }
 
+template <typename CMType>
+void output_single_model(VW::workspace& all, automl<CMType>& data)
+{
+  options_i& options = *all.options;
+  if (!options.was_supplied("predict_only_model")) { return; }
+  // Clear non-champ weights first
+
+  std::swap(*data.cm->_gd_normalized, data.cm->per_live_model_state_double[0]);
+  std::swap(*data.cm->_gd_total_weight, data.cm->per_live_model_state_double[1]);
+  std::swap(*data.cm->_sd_gravity, data.cm->per_live_model_state_double[2]);
+  std::swap(*data.cm->_cb_adf_event_sum, data.cm->per_live_model_state_uint64[0]);
+  std::swap(*data.cm->_cb_adf_action_sum, data.cm->per_live_model_state_uint64[1]);
+
+  clear_non_champ_weights(data.cm->weights, data.cm->estimators.size(), data.cm->wpp);
+
+  uint64_t multiplier = static_cast<uint64_t>(data.cm->wpp) << data.cm->weights.stride_shift();
+  for (uint32_t index = 0; index < data.cm->weights.mask(); index += multiplier)
+  {
+    if (data.cm->weights[index] != 0.0f)
+    {
+      uint32_t cb_ind = index / data.cm->wpp;
+      for (uint32_t stride = 0; stride < (static_cast<uint32_t>(1) << data.cm->weights.stride_shift()); ++stride)
+      { std::swap(data.cm->weights[index + stride], data.cm->weights[cb_ind + stride]); }
+    }
+  }
+  uint32_t num_bits =
+      options.was_supplied("bit_precision") ? options.get_typed_option<uint32_t>("bit_precision").value() : 18;
+  options.remove_option("automl");
```

Comment:
do you have to remove all the other automl related options? if not when re-loading you'll get a warning/error about using un-used args. Try it out with --priority_challenger or something like that.

Reply:
you might be able to automate it similar to how we extract the name of a reduction setup, but instead of registering the name you register all the options https://github.com/VowpalWabbit/vowpal_wabbit/blob/417012d772bf26cecc8d234a820994aa65144d82/vowpalwabbit/core/src/reduction_stack.cc#L113

Reply:
updated to check and remove all automl options